R/utilities.R

```diff
@@ -204,6 +204,102 @@ match_output_arguments <- function(input_args = c(),
   return(output_args)
 }
 
+# Safe Directory Create
+# Used to create a directory for the functions
+
+safe_dir_create <- function(dir_name){
+  dir_of_interest <- file.path(dir_name)
+  if(dir.exists(dir_of_interest)){
+    invisible()
+  } else {
+    dir.create(dir_of_interest, recursive = TRUE)
+  }
+}
+
+# Safe Copy Stan Model
+# This copies over all of the associated functions and stan files
+# when the user users the internal model. To be used as internal function only.
+# Users shouldn't need to access this directly.
+
+safe_file_copy <- function(files, x){
+  tryCatch(file.copy(files, x, overwrite = FALSE),
+           error = message("File Exists, Using Existing File\n"))
+}
+
+copy_models <- function(dir_path){
+  # First copy functions
+  app_file_loc <- system.file(file.path("stan", "functions"), package = "EpiNow2")
+  
+  safe_dir_create(file.path(dir_path, "functions"))
+  
+  functions_to_copy <- list.files(app_file_loc, full.names = TRUE)
+  
+  
+  
+  lapply(functions_to_copy, safe_file_copy, x = file.path(dir_path, "functions"))
+  
+  # Copy Models
+  stan_file_loc <- system.file(file.path("stan", "estimate_infections.stan"), 
+                               package = "EpiNow2")
+  
+  if(!file.exists(file.path(dir_path,
+                       "estimate_infections.stan"))){
+    tryCatch(R.utils::copyFile(stan_file_loc, file.path(dir_path,
+                                                        "estimate_infections.stan"), overwrite = FALSE),
+             error = message("Model already exists"))
+  }
+  
+}
+
+
+# Helper function for Formatting List for CmdStanR
+make_cmdstan_list <-function(stan_args){
+  
+  #mk_init <- list(stan_args$init(),stan_args$init(),stan_args$init(),stan_args$init())
+  mk_init <- stan_args$init
+  
+  
+  
+  out <- list(
```

Comment:
This should respect the user specified Stan arguments via `stan_args`. Check [this commit](https://github.com/hsbadr/EpiNow2/commit/ccee8f08908279a60891aa89639ae3e3f54a3b44)!